src/addons/sourcemod/scripting/zr/weapons/zmarket.inc

```diff
@@ -156,41 +153,28 @@ public Action ZMarketBuyCommand(int client, int argc)
     {
         WeaponsGetZMarketCommand(weaponindex, zmarketcommand, sizeof(zmarketcommand));
 
-        if (FindCharInString(zmarketcommand, ',') != -1)
-        {
-            int idx;
-            int lastidx;
-            while ((idx = FindCharInString(zmarketcommand[lastidx], ',')) != -1)
-            {
-                if (!strncmp(command, zmarketcommand[lastidx], idx))
-                {
-                    WeaponsGetName(weaponindex, weaponname, sizeof(weaponname));
-                    ZMarketEquip(client, weaponname);
-                    return Plugin_Handled;
-                }
-
-                lastidx += ++idx;
+        // Split the commands and check each one
+        char commands[8][32];
+        int count = ExplodeString(zmarketcommand, ",", commands, sizeof(commands), sizeof(commands[]));
 
-                if (FindCharInString(zmarketcommand[lastidx], ',') == -1 && zmarketcommand[lastidx+1] != '\0')
-                {
-                    if (!strncmp(command, zmarketcommand[lastidx], idx))
-                    {
-                        WeaponsGetName(weaponindex, weaponname, sizeof(weaponname));
-                        ZMarketEquip(client, weaponname);
-                        return Plugin_Handled;
-                    }
-                }
-            }
-        }
-        else
+        for (int i = 0; i < count; i++)
         {
-            if(strcmp(command, zmarketcommand, false) == 0)
+            TrimString(commands[i]);
+            if (strcmp(command, commands[i], false) == 0)
             {
                 WeaponsGetName(weaponindex, weaponname, sizeof(weaponname));
                 ZMarketEquip(client, weaponname);
                 return Plugin_Handled;
             }
         }
+
+        // Also check for single command case
+        if (strcmp(command, zmarketcommand, false) == 0)
```

Comment:
instead of duplicating code, you could do check before the loop if count <= 0 and put command in commands[0] so that way we have only one loop with important functions